src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs

```diff
@@ -55,16 +57,24 @@ internal class ConnectionManager : IConnectionManager {
             interactiveWorkflow.RSession.Connected += RSessionOnConnected;
             interactiveWorkflow.RSession.Disconnected += RSessionOnDisconnected;
 
-            _shell.DispatchOnUIThread(() => _disposableBag.Add(_statusBar.AddItem(new ConnectionStatusBar { DataContext = _statusBarViewModel })));
+            _shell.Idle += OnIdle;
 
             // Get initial values
-            var userConnections = GetConnectionsFromSettings();
+            var userConnections = CreateConnectionList();
             _userConnections = new ConcurrentDictionary<Uri, IConnection>(userConnections);
 
             UpdateRecentConnections();
             SwitchBrokerToLastConnection();
         }
 
+        private void OnIdle(object sender, EventArgs e) {
+            _shell.Idle -= OnIdle;
+
+            _disposableBag.Add(_statusBar.AddItem(new ConnectionStatusBar {
```

Comment:
It is, the problem was with `DispatchOnUIThread` that causes message pump. I can plain call `AddItem` if you think ctor is only called on the main thread.